mongodbatlas/fw_data_source_mongodbatlas_search_deployment.go

```diff
@@ -0,0 +1,99 @@
+package mongodbatlas
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ datasource.DataSource = &SearchDeploymentDS{}
+var _ datasource.DataSourceWithConfigure = &SearchDeploymentDS{}
+
+func NewSearchDeploymentDS() datasource.DataSource {
+	return &SearchDeploymentDS{
+		DSCommon: DSCommon{
+			dataSourceName: searchDeploymentName,
+		},
+	}
+}
+
+type tfSearchDeploymentDSModel struct {
+	ID          types.String `tfsdk:"id"`
+	ClusterName types.String `tfsdk:"cluster_name"`
+	ProjectID   types.String `tfsdk:"project_id"`
+	Specs       types.List   `tfsdk:"specs"`
+	StateName   types.String `tfsdk:"state_name"`
+}
+
+type SearchDeploymentDS struct {
+	DSCommon
+}
+
+func (d *SearchDeploymentDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
```

Comment:
this is more of a generic question: in the context of a data source I don't understand the concept of `Computed`: aren't they all computed?

On the other side I'd imagine `Required` means that if we don't fill it up with the backend response terraform would raise an error.

Reply:
> in the context of a data source, aren't they all computed?

Not exactly. When using a data source there are some attributes that need to be provided by the user in order to identify the resource that will be fetched. In this case the resource is identified through the `project_id` and `cluster_name`, which is why they are marked as required.
You will see this is also clarified in the documentation of the data source, separating the concept of arguments (attributes that have to be defined), and then an attributes sections for values that are exported (or as we call computed).